### ISSUE TITLE
Add NULL support in count function

### ISSUE THREAD

**Issue by yupeng9** [apache/pinot#7543]
To support queries like `count(col)` where `col` can have NULL values, or  `count(CASE WHEN condition THEN v END)`

In certain cases, there's no workaround to calculating such values.

**Comment by Jackie-Jiang** [comment]
Please take a look at the feature proposed by @atris #7519. We can model the query as `SELECT COUNT(*) FILTER (WHERE col IS NOT NULL), ...`

**Comment by yupeng9** [comment]
Yes, I left some comments in the doc. At the syntax level, can we make the feature SQL standard compatible?
